### PR TITLE
Move scenario app to use iPhone Se 3rd gen iOS 16.2, un-skip test `TwoPlatformViewsWithOtherBackDropFilterTests`

### PR DESCRIPTION
**Pull request by cyanglaz**
iPhone 8 is old and not installed by default with the latest XCodes. Move the scenario test to a newer iPhone that has a home button. (Unfortunately, iPhones without home button have the bottom navigation bar that animates, resulting inconsistency during screenshots)

Fixes https://github.com/flutter/flutter/issues/125167

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt. See [testing the engine] for instructions on writing and running engine tests.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/main/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by cyanglaz** [comment]
This is the image generated from my local arm m1 machine for `TwoPlatformViewsWithOtherBackDropFilterTests`  compares to the golden image. 
It looks like a bug. I will look into this tomorrow. 
golden_two_platform_views_with_other_backdrop_filter_iPhone SE (3rd generation)_16_1_4824D849-B7E0-4531-ACE5-D421ABE67C31 2_simulator

File: testing/scenario_app/ios/Scenarios/ScenariosUITests/golden_two_platform_views_with_other_backdrop_filter_iPhone SE (3rd generation)_16.2_simulator.png
Comment (cyanglaz) [inline testing/scenario_app/ios/Scenarios/ScenariosUITests/golden_two_platform_views_with_other_backdrop_filter_iPhone SE (3rd generation)_16.2_simulator.png:1]: This is the screenshot generated on Arm, its significantly different:
image

Comment (cyanglaz) [reply testing/scenario_app/ios/Scenarios/ScenariosUITests/golden_two_platform_views_with_other_backdrop_filter_iPhone SE (3rd generation)_16.2_simulator.png:1]: @jason-simmons Does it look the same with the one you have it on your local.

**Comment by cyanglaz** [comment]
Well, my local arm failed the `TwoPlatformViewsWithOtherBackDropFilterTests` but it seems passed on ci arm machine? I'm ok with it as long as it passes on ci: https://ci.chromium.org/ui/p/flutter/builders/try/Linux%20Engine%20Drone/786971/overview

@jmagman

File: testing/scenario_app/ios/Scenarios/ScenariosUITests/golden_platform_view_cliprect_after_moved_iPhone SE (3rd generation)_16.2_simulator.png
Comment (cyanglaz) [inline testing/scenario_app/ios/Scenarios/ScenariosUITests/golden_platform_view_cliprect_after_moved_iPhone SE (3rd generation)_16.2_simulator.png:1]: I think git is confused and compared the wrong file. You can check the full file names by hovering the mouse to file name above. Same as the one below.

**Review by jmagman — APPROVED** [review]
LGTM!

**Review by jmagman — COMMENTED** [review]
You can land this now, rebase https://github.com/flutter/engine/pull/41640 onto it, and either update `ios-16-0_14a5294e` to `ios-16-2_14c18` or remove totally and update the images again.

**Pull request merged**